mail_activity_default_assignee/wizard/mail_activity_schedule.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2024 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class MailActivitySchedule(models.TransientModel):
+    _inherit = "mail.activity.schedule"
+
+    @api.depends("activity_type_id")
+    def _compute_activity_user_id(self):
+        res = super()._compute_activity_user_id()
+        for scheduler in self.filtered(
+            lambda rec: not rec.activity_type_id.default_user_id
+            and rec.activity_type_id.default_user_field_id
+        ):
+            fname = scheduler.activity_type_id.default_user_field_id.name
+            res_model = self.env[scheduler.res_model]
+            user_ids = scheduler._get_user_ids(res_model, fname)
+            if not user_ids:
+                continue
+            scheduler.activity_user_id = user_ids[0]
```

Comment:
I think you can simplify this bit by reusing some internal methods.
You can drop the `_get_user_ids`, too.. 

```suggestion
            if users := scheduler._get_applied_on_records().mapped(fname):
                scheduler.activity_user_id = users[:1]
```

ℹ️ requirement: add `@api.depends('res_ids')`